### PR TITLE
Added NumpyArrayNodeCostRecorder

### PR DESCRIPTION
**Pull request by jetuk**

### CODE REVIEW

File: tests/test_recorders.py [hunk 1]
```diff
@@ -162,11 +164,15 @@ def test_numpy_recorder(simple_linear_model):
...
     assert rec.data.shape == (365, 1)
+    assert cost_rec.data.shape == (365, 1)
     assert np.all((rec.data - 10.0) < 1e-12)
+    np.testing.assert_allclose(cost_rec.data, np.ones_like(cost_rec.data) * -2.0)
 
     df = rec.to_dataframe()
+    df2 = cost_rec.to_dataframe()
     assert df.shape == (365, 1)
     assert np.all((df.values - 10.0) < 1e-12)
+    np.testing.assert_allclose(df2.values, np.ones_like(cost_rec.data) * -2.0)
```
Comment (Batch21) [inline tests/test_recorders.py:175]: Is it worth having a test where the node cost varies?

**Review by Batch21 — CHANGES_REQUESTED** [review]
Looks good - surprised this doesn't exist already. Just a query on whether an additional test is necessary.

File: tests/test_recorders.py (hunk 1 above)
Comment (jetuk) [reply tests/test_recorders.py]: Done.

**Review by Batch21 — APPROVED** [review]
Tests look good, thanks!

**Pull request merged**